pkg/stub/handler.go

```diff
@@ -26,16 +27,56 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	case *v1alpha1.Jaeger:
 		if event.Deleted {
 			logrus.Infof("Deleting '%s'", o.Name)
-			// aparently, everything created by the CR should cascade the deletion to the
-			// resources it created, so, no need to do anything
-			// on a next version, we could think about cleaning up the data pertaining to
-			// us at the storage level, but not for now
 			return nil
 		}
 
+		// we need a name!
+		if o.Name == "" {
+			logrus.Infof("This Jaeger instance was created without a name. Setting it to 'my-jaeger'")
+			o.Name = "my-jaeger"
+		}
+
+		// normalize the storage type
+		if o.Spec.Storage.Type == "" {
+			logrus.Infof("Storage type wasn't provided for the Jaeger instance '%v'. Falling back to 'memory'", o.Name)
+			o.Spec.Storage.Type = "memory"
+		}
+
+		if unknownStorage(o.Spec.Storage.Type) {
+			logrus.Infof(
+				"The provided storage type for the Jaeger instance '%v' is unknown ('%v'). Falling back to 'memory'",
+				o.Name,
+				o.Spec.Storage.Type,
+			)
+			o.Spec.Storage.Type = "memory"
+		}
+
+		// normalize the deployment strategy
+		if strings.ToLower(o.Spec.Strategy) != "production" {
```

Comment:
does this operator support deploying all-in-one with real storage?

Reply:
Yes